loopout.go

```diff
@@ -635,12 +709,75 @@ func (s *loopOutSwap) payInvoiceAsync(ctx context.Context,
 	}
 
 	// Lookup state of the swap payment.
-	paymentStateCtx, cancel := context.WithCancel(ctx)
+	payCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	payStatusChan, payErrChan, err := s.lnd.Router.SendPayment(
-		paymentStateCtx, req,
+	start := time.Now()
+	paymentStatus, attempts, err := s.sendPaymentWithRetry(
+		payCtx, hash, &req, maxRetries, routingPlugin, pluginType,
 	)
+
+	dt := time.Since(start)
+	paymentSuccess := err == nil &&
+		paymentStatus.State == lnrpc.Payment_SUCCEEDED
+
+	if err := s.swapKit.server.ReportRoutingResult(
+		ctx, s.swapInfo().SwapHash, s.swapInvoicePaymentAddr, pluginType,
+		paymentSuccess, int32(attempts), dt.Milliseconds(),
+	); err != nil {
+		s.log.Warnf("Failed to report routing result: %v", err)
+	}
+
+	return paymentStatus, err
+}
+
+// sendPaymentWithRetry will send the payment, optionally with the passed
+// routing plugin retrying at most maxRetries times.
+func (s *loopOutSwap) sendPaymentWithRetry(ctx context.Context,
+	hash lntypes.Hash, req *lndclient.SendPaymentRequest, maxRetries int,
+	routingPlugin RoutingPlugin, pluginType RoutingPluginType) (
```

Comment:
I think the `routingPlugin` already contains the `pluginType` info since that is how it's initialized. We can add a method to that interface returning us the plugin type to remove the redundancy here.